modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/classloader/net/MuleUrlStreamHandlerFactory.java

```diff
@@ -87,10 +87,17 @@ private URLStreamHandler defaultHandlerCreateStrategy(String protocol) {
     while (handler == null && tokenizer.hasMoreTokens()) {
       String packagePrefix = tokenizer.nextToken().trim();
       String className = packagePrefix + "." + protocol + ".Handler";
-      try {
-        handler = (URLStreamHandler) ClassUtils.instantiateClass(className);
-      } catch (Exception ex) {
-        // not much we can do here
+
+      // MULE-15867 - MG says: for some mysterious reason, Java 11 is fine instantiating this class explicitly but logs
+      // a warning when instantiated by reflection.
+      if ("sun.net.www.protocol.jar.Handler".equals(className)) {
+        handler = new sun.net.www.protocol.jar.Handler();
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Use classes from the Java API instead of Sun classes. ([squid:S1191](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1191))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7616&issues=AWhigVjkotIXnmsUYdyO&open=AWhigVjkotIXnmsUYdyO)

[ISSUE_KEY]: AWhigVjkotIXnmsUYdyO